src/v2/controllers/vreplicaset_controller/proof/helper_invariants/predicate.rs

```diff
@@ -175,30 +172,21 @@ pub open spec fn every_delete_matching_pod_request_implies_at_after_delete_pod_s
     |s: ClusterState| {
         forall |msg: Message| #![trigger s.in_flight().contains(msg)] {
             let content = msg.content;
-            let req = content.get_delete_request();
-            let key = content.get_delete_request().key;
+            let req = content.get_get_then_delete_request();
+            let key = content.get_get_then_delete_request().key;
             let obj = s.resources()[key];
             &&& s.in_flight().contains(msg)
             &&& msg.src.is_Controller()
             &&& msg.src.get_Controller_0() == controller_id
             &&& msg.dst.is_APIServer()
             &&& msg.content.is_APIRequest()
-            &&& content.is_delete_request()
+            &&& content.is_get_then_delete_request()
             &&& s.resources().contains_key(key)
             &&& owned_selector_match_is(vrs, obj)
-            // NOTE: We require that the resource version in etcd is
```

Comment:
What we probably need is another invariant stating that no 'delete' requests are issued on our pods.